cli/pack/common.go

```diff
@@ -0,0 +1,510 @@
+	envPath,
+	envBinPath,
+	envModulesPath,
+}
+
+var defaultBlackListExpressions = []string{
+	"\\w+\\/.+", // stops to watching apps inside the directories
+	"^\\.\\w+",  // stops from indexing files starting with . like .rocks
+	"^\\.$",
+	"\\w+.yaml",
+	"\\w+.tar.gz",
+}
+
+func pack(ctx *PackCtx, cmdCtx *cmdcontext.CmdCtx) (string, error) {
+	var err error
+
+	startPath, _ := os.Getwd()
+	defer func() {
+		_ = os.Chdir(startPath)
+	}()
+	err = goToEnvRoot(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	// create temporary directory step
+	basePath, err := os.MkdirTemp("", "tt_pack")
+	if err != nil {
+		return "", err
+	}
+
+	defer func() {
+		if err != nil {
+			_ = os.RemoveAll(basePath)
+		}
+	}()
+
+	err = createPackageStructure(basePath)
+	if err != nil {
+		return "", err
+	}
+
+	// copy binaries step
+	if ctx.App.BinDir != "" &&
+		((!ctx.TarantoolIsSystem && !ctx.WithoutBinaries) ||
+			ctx.WithBinaries) && !ctx.RpmDeb.WithTarantoolDeps {
+		baseEnvBinPath := filepath.Join(basePath, envPath, binPath)
+		err = copy.Copy(ctx.App.BinDir, baseEnvBinPath)
+		if err != nil {
+			return "", err
+		}
+	}
+
+	// copy modules step
+	if ctx.ModulesDirectory != "" {
+		baseEnvModulesPath := filepath.Join(basePath, envPath, modulesPath)
+		err = copy.Copy(ctx.ModulesDirectory, baseEnvModulesPath)
+		if err != nil {
+			return "", err
+		}
+	}
+
+	// collect app list step
+	appList := ctx.AppList
+	if ctx.AppList == nil && ctx.App.InstancesEnabled != "." {
+		// initialize a default list of strings for preparing a black list
+		blackListExpressions := prepareDefaultBlackListExpressions()
+
+		blackList, err := prepareBlackList(blackListExpressions)
+		if err != nil {
+			return "", fmt.Errorf("failed to compile regular expressions: %v", err)
+		}
+
+		appList, err = collectAppList(ctx.App.InstancesEnabled, blackList)
+		if err != nil {
+			return "", err
+		}
+	} else if ctx.App.InstancesEnabled == "." {
+		// if running tt pack with instances_enabled: .
+		// it supposed to pack a current directory as an application
+		appPath, err := os.Getwd()
+		if err != nil {
+			return "", err
+		}
+		appList = []string{filepath.Base(appPath)}
+		err = os.Chdir("..")
+		if err != nil {
+			return "", err
+		}
+	}
+	log.Infof("Apps to pack: %v", appList)
+
+	// copy all apps to a temp directory step
+	for _, appName := range appList {
+		err = copyAppSrc(ctx, appName, basePath)
+		if err != nil {
+			return "", err
+		}
+
+		if ctx.Archive.All {
+			err = copyArtifacts(ctx, appName, basePath)
+			if err != nil {
+				return "", err
+			}
+		}
+
+		err = createAppSymlink(ctx, appName, basePath)
+		if err != nil {
+			return "", err
+		}
+
+		err = buildRocks(cmdCtx, appName, basePath)
+		if err != nil && !os.IsNotExist(err) {
+			return "", err
+		}
+	}
+
+	err = createEnv(ctx, basePath)
+	if err != nil {
+		return "", err
+	}
+	return basePath, nil
+}
+
+func createPackageStructure(destPath string) error {
+	baseVarPath := filepath.Join(destPath, varPath)
```

Comment:
I see a lot of similar code for paths generation. Pls, extract this stuff to separate function, which accepts base directory and return a structure with all members set.
Or a function which accepts base directory and returns a closure, which can generate required paths.